flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/SinkCodeGenerator.scala

```diff
@@ -223,10 +224,10 @@ object SinkCodeGenerator {
           case (fieldTypeInfo, i) =>
             val requestedTypeInfo = tt.getTypeAt(i)
             validateFieldType(requestedTypeInfo)
-            if (!areTypesCompatible(
-              fromTypeInfoToLogicalType(fieldTypeInfo),
-              fromTypeInfoToLogicalType(requestedTypeInfo)) &&
-              !requestedTypeInfo.isInstanceOf[GenericTypeInfo[Object]]) {
+            if (!PlannerTypeUtils.isAssignable(
+                fromTypeInfoToLogicalType(fieldTypeInfo),
+                fromTypeInfoToLogicalType(requestedTypeInfo)) &&
+                !requestedTypeInfo.isInstanceOf[GenericTypeInfo[Object]]) {
```

Comment:
Should the type be assignable or just equals strictly ? I think it is the later.

Reply:
1. It's safe to be assignable (ignore precision and nullablitiy).
2. Currently, almost all the sinks are using TypeInformations (e.g. UpsertTableSink can only use TypeInformations). And TypeInformations will lose precision informations which will lead to be always type-incompatible.

Reply:
It can be discussed to whether it should be replaced by `isInteroperable`.
But never `just equals strictly`, it is too strict.